Chatkit/Redux/Actions/Action.swift

```diff
@@ -0,0 +1,2 @@
+
+class Action {}
```

Comment:
Is the reason `Action` has become a class because it wasn't possible to specify a `typealias ActionType = Action` when using a protocol?

Reply:
Because there is **a lot** baggage that comes with this alteration.

Reply:
This approach is my preference (see last commit on this branch)
https://github.com/pusher/chatkit-swift/tree/OKP/Reducers-Action-not-class

Why/How: 
- `Action` is still a protocol, we've added a `MasterAction` enum
- No reference types floating around
- No need to define custom `Equatable` implementations for every `Action`
- `MasterAction` enum prevents necessity to cast anything
- Its now possible to perform `XCTAssertEqual(masterAction, anotherMasterAction)`

Reply:
Yes, this is due to the mentioned `typealias`.

`MasterAction` has one big downside which I would prefer to avoid. It does not allow to constrain input parameters to a specific subtype of `Action`.

Reply:
I think it does.  If you check my branch you'll see that `UserSubscription` reducers all use constrained `Action`s, for example...
```
typealias ActionType = ReadStateUpdatedAction 
```
https://github.com/pusher/chatkit-swift/blob/8127836165fd9f06b8d2d445675ee9b5c90ecf2f/Chatkit/Redux/Reducers/User%20Subscription/ReadStateUpdatedReducer.swift#L8

`MasterReducer` and `Model` reducers use `MasterAction`.

Reply:
This approach means that we introduce subactions.